File: wirtualny_swiat/src/swiat/Organizm.java
package swiat;
import java.awt.Color;

public abstract class Organizm {
    private String nazwa;
    private char symbol;
    private int sila;
    private int inicjatywa;
    private int wiek;
    private int polozenieX;
    private int polozenieY;
    private boolean aktywnaNiesmiertelnosc;
    private boolean czyZyje;
    public final int RUCH_NIEAKTYWNY = -1;
    public final int RUCH_PRAWO = 1;
    public final int RUCH_LEWO = 2;
    public final int RUCH_GORA = 3;
    public final int RUCH_DOL = 4;
    public abstract void akcja(Swiat swiat);
    public abstract void kolizja(Swiat swiat, Organizm atakujacy);
    public abstract Color rysowanie();
    public abstract Organizm stworzNowySklonowanyObiekt();


    public void kopiujObiekt(final Organizm innyOrganizm) {
        this.nazwa = innyOrganizm.getNazwa();
        this.symbol = innyOrganizm.getSymbol();
        this.sila = innyOrganizm.getSila();
        this.inicjatywa = innyOrganizm.getInicjatywa();
        this.wiek = innyOrganizm.getWiek();
        this.polozenieX = innyOrganizm.getPolozenieX();
        this.polozenieY = innyOrganizm.getPolozenieY();
    }


    public boolean czyOdbilAtak(Organizm atakujacy, Organizm atakowany) {
        int silaAtakowanego = atakowany.getSila();
        int silaAtakujacego = atakujacy.getSila();

        return silaAtakujacego < silaAtakowanego;
    }


    public String getNazwa() {
        return nazwa;
    }


    public char getSymbol() {
        return symbol;
    }


    public int getSila() {
        return sila;
    }


    public int getInicjatywa() {
        return inicjatywa;
    }


    public int getWiek() {
        return wiek;
    }


    public  int getPolozenieX() {
        return polozenieX;
    }


    public  int getPolozenieY() {
        return polozenieY;
    }


    public boolean czyNiesmiertelny() {
        return aktywnaNiesmiertelnosc;
    }


    public boolean getCzyZyje() {
        return czyZyje;
    }


    public void setNazwa(String nazwa) {
        this.nazwa = nazwa;
    }


    public void setSymbol(char symbol) {
        this.symbol = symbol;
    }


    public void setSila(int sila) {
        this.sila = sila;
    }


    public void setInicjatywa(int inicjatywa) {
        this.inicjatywa = inicjatywa;
    }


    public  void setWiek(int wiek) {
        this.wiek = wiek;
    }


    public void setCzyZyje(boolean wyrazenie) {
        this.czyZyje = wyrazenie;
    }


    public void setPolozenieX(int polozenieX) {
        this.polozenieX = polozenieX;
    }


    public void setPolozenieY(int polozenieY) {
        this.polozenieY = polozenieY;
    }


    public void setNiesmiertelnosc(boolean wyrazenie) {
        this.aktywnaNiesmiertelnosc = wyrazenie;
    }


    public static String wypiszPolozenie(final int polozenieX, final int polozenieY) {
        return " ("+polozenieX+","+polozenieY+")";
    }
}

File: wirtualny_swiat_python/Organizm.py
from abc import ABC, abstractmethod
from Swiat import Swiat


class Organizm(ABC):
    RUCH_NIEAKTYWNY = -1
    RUCH_PRAWO = 1
    RUCH_LEWO = 2
    RUCH_GORA = 3
    RUCH_DOL = 4

    def __init__(self, x, y, wiek):
        self.nazwa = ""
        self.symbol = ""
        self.sila = 0
        self.inicjatywa = 0
        self.wiek = wiek
        self.polozenieX = x
        self.polozenieY = y
        self.aktywnaNiesmiertelnosc = False
        self.czyZyje = False

    @staticmethod
    def wypiszPolozenie(x, y):
        return " ({},{})".format(x, y)

    @staticmethod
    def czyOdbilAtak(atakujacy, atakowany):
        silaAtakowanego = atakowany.getSila()
        silaAtakujacego = atakujacy.getSila
        return silaAtakujacego < silaAtakowanego

    @abstractmethod
    def akcja(self, swiat):
        pass

    @abstractmethod
    def kolizja(self, swiat, atakujacy):
        pass

    @abstractmethod
    def rysowanie(self):
        pass

    @abstractmethod
    def stworzNowySklonowowanyObiekt(self):
        pass

    def kopiujObiekt(self, innyOrganizm):
        self.nazwa = innyOrganizm.getNazwa()
        self.symbol = innyOrganizm.getSymbol()
        self.sila = innyOrganizm.getSila()
        self.inicjatywa = innyOrganizm.getInicjatywa()
        self.wiek = innyOrganizm.getWiek()
        self.polozenieX = innyOrganizm.getPolozenieX()
        self.polozenieY = innyOrganizm.getPolozenieY()

    def getNazwa(self):
        return self.nazwa

    def getSymbol(self):
        return self.symbol

    def getSila(self):
        return self.sila

    def getInicjatywa(self):
        return self.inicjatywa

    def getWiek(self):
        return self.wiek

    def getPolozenieX(self):
        return self.polozenieX

    def getPolozenieY(self):
        return self.polozenieY

    def getCzyZyje(self):
        return self.czyZyje

    def czyNiesmiertelny(self):
        return self.aktywnaNiesmiertelnosc

    def setNazwa(self, nazwa):
        self.nazwa = nazwa

    def setSymbol(self, symbol):
        self.symbol = symbol

    def setSila(self, sila):
        self.sila = sila

    def setInicjatywa(self, inicjatywa):
        self.inicjatywa = inicjatywa

    def setWiek(self, wiek):
        self.wiek = wiek

    def setPolozenieX(self, x):
        self.polozenieX = x

    def setPolozenieY(self, y):
        self.polozenieY = y

    def setCzyZyje(self, wyrazenie):
        self.czyZyje = wyrazenie

    def setNiesmiertelnosc(self, wyrazenie):
        self.aktywnaNiesmiertelnosc = wyrazenie
